File: 品味Java的21種設計模式/dp_cpp/decorator/example2/C3.cpp
#include "C3.h"

namespace cn {
namespace javass {
namespace dp {
namespace decorator {
namespace example2 {

void C3::a1() {
    //转调A对象的功能
    a->a1();
}

void C3::b1() {
    //转调B对象的功能
    b->b1();
}

void C3::c11() {
    puts("now in C3.c11");
}
}
}
}
}
}

File: 品味Java的21種設計模式/dp_java/decorator/example2/C3.java
package cn.javass.dp.decorator.example2;

public class C3 {
    private A a = new A();
    private B b = new B();

    public void a1() {
        //转调A对象的功能
        a.a1();
    }
    public void b1() {
        //转调B对象的功能
        b.b1();
    }

    public void c11() {
        System.out.println("now in C3.c11");
    }
}
